feature/system/src/main/java/io/github/droidkaigi/confsched2020/system/ui/viewmodel/SystemViewModel.kt

```diff
@@ -35,4 +36,20 @@ class SystemViewModel @Inject constructor(
         // launch a Custom Tabs Activity
         customTabsIntent.launchUrl(activity, Uri.parse(url))
     }
+
+
+    fun openCalendar(
+        title: String,
+        location: String,
+        startUnixMillis: Long,
+        endUnixMillis: Long
+    ) {
+        val intent = Intent(Intent.ACTION_INSERT)
+            .setData(CalendarContract.Events.CONTENT_URI)
+            .putExtra(CalendarContract.EXTRA_EVENT_BEGIN_TIME, startUnixMillis)
+            .putExtra(CalendarContract.EXTRA_EVENT_END_TIME, endUnixMillis)
+            .putExtra(CalendarContract.Events.TITLE, "DroidKaigi2020: $title")
+            .putExtra(CalendarContract.Events.EVENT_LOCATION, location)
+        activity.startActivity(intent)
```

Comment:
@takahirom 
I think we should handle it somehow if a device does not have any Calendar application. However, we may need to consider the design of handling such an exception. Can I make it a future work and accept this change as it is now?

Reply:
It is better to try-catch when starting implicit Intent. 👍